src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -263,7 +263,10 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 : context.SemanticModel.LookupSymbols(context.LeftToken.SpanStart);
 
             // Filter out any extension methods that might be imported by a using static directive.
-            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod());
+            // But include extension methods declared in the context's type
+            var contextOuterTypes = context.GetOuterTypes(true, cancellationToken);
+            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod()
+                || contextOuterTypes.Any(x => x.CompareTo(symbol.ContainingType) == 0));
```

Comment:
If the main concern is repeatedly computing the enclosing type + outer type list, just compute it once into a local outside your lambda. I wouldn't worry too much about the case where you compute that list and then have no extension methods: as you've seen the context already memoizes the outer types list.